src/test/run-make-fulldeps/coverage-reports/Makefile

```diff
@@ -98,28 +98,29 @@ endif
 			)
 
 	# Run it through rustdoc as well to cover doctests
-	LLVM_PROFILE_FILE="$(TMPDIR)"/$@-%p.profraw \
+	LLVM_PROFILE_FILE="$(TMPDIR)"/$@-%p-%m.profraw \
 			$(RUSTDOC) --crate-name workaround_for_79771 --test $(SOURCEDIR)/$@.rs \
 			$$( sed -n 's/^\/\/ compile-flags: \([^#]*\).*/\1/p' $(SOURCEDIR)/$@.rs ) \
 			-L "$(TMPDIR)" -Zinstrument-coverage \
 			-Z unstable-options --persist-doctests=$(TMPDIR)/rustdoc-$@
 
 	# Postprocess the profiling data so it can be used by the llvm-cov tool
 	"$(LLVM_BIN_DIR)"/llvm-profdata merge --sparse \
-			"$(TMPDIR)"/$@-*.profraw \
+			--num-threads=1 \
```

Comment:
Based on the documentation, and just googling for more info on the `--num-threads` option to both `llvm-profdata` and `llvm-cov`, I don't see how this option would have any impact.

For example, https://github.com/llvm/llvm-project/blob/main/llvm/test/tools/llvm-cov/multithreaded-report.test seems to imply the option should have no effect on the results.

If you have evidence of some undocumented side effect of adding this option, please add a comment in the Makefile explaining the side effect you expect, and a link to the online reference (documentation or code) that shows the evidence of this side effect. 

Otherwise, I don't think this option should be added, in either command.

I